File: pytools/data/data.code.golang.client/msg_test_send.go
package proto

import (
	"packet"
)

type MsgTestSend struct {
	idU8                     uint8
	roleBase                 *MsgRoleBase
	idF32                    []float32
	idOpU8Flag               uint8
	idOpU8                   uint8
	opRoleBaseFlag           uint8
	opRoleBase               *MsgRoleBase
}

func (this *MsgTestSend) Encode() []byte {
	pack := packet.NewWriteBuff(64)

	pack.WriteUint8(this.idU8)
	pack.WriteBytes(this.roleBase.Encode())
	idF32Count := uint16(len(this.idF32))
	pack.WriteUint16(idF32Count)
	for i := uint16(0); i < idF32Count; i++ {
		pack.WriteFloat32(this.idF32[i])
	}
	pack.WriteUint8(this.idOpU8Flag)
	if this.idOpU8Flag == 1 {
		pack.WriteUint8(this.idOpU8)
	}
	pack.WriteUint8(this.opRoleBaseFlag)
	if this.opRoleBaseFlag == 1 {
		pack.WriteBytes(this.opRoleBase.Encode())
	}

	return pack.ReadBytes()
}

func MsgTestSendDecode(pack *packet.Packet) *MsgTestSend {
	msgTestSend := &MsgTestSend{}

	msgTestSend.idU8 = pack.ReadUint8()
	msgTestSend.roleBase = MsgRoleBaseDecode(pack)
	idF32Count := pack.ReadUint16()
	for ;idF32Count > 0; idF32Count-- {
		msgTestSend.idF32 = append(msgTestSend.idF32, pack.ReadFloat32())
	}
	msgTestSend.idOpU8Flag = pack.ReadUint8()
	if msgTestSend.idOpU8Flag == 1 {
		msgTestSend.idOpU8 = pack.ReadUint8()
	}
	msgTestSend.opRoleBaseFlag = pack.ReadUint8()
	if msgTestSend.opRoleBaseFlag == 1 {
		msgTestSend.opRoleBase = MsgRoleBaseDecode(pack)
	}
	return msgTestSend
}

func (this *MsgTestSend) SetIdU8(idU8 uint8) {
	this.idU8 = idU8
}

func (this *MsgTestSend) GetIdU8() uint8 {
	return this.idU8
}

func (this *MsgTestSend) SetRoleBase(roleBase *MsgRoleBase) {
	this.roleBase = roleBase
}

func (this *MsgTestSend) GetRoleBase() *MsgRoleBase {
	return this.roleBase
}

func (this *MsgTestSend) SetIdF32(idF32 []float32) {
	this.idF32 = idF32
}

func (this *MsgTestSend) GetIdF32() []float32 {
	return this.idF32
}

func (this *MsgTestSend) SetIdOpU8(idOpU8 uint8) {
	this.idOpU8Flag = 1
	this.idOpU8 = idOpU8
}

func (this *MsgTestSend) GetIdOpU8() uint8 {
	return this.idOpU8
}

func (this *MsgTestSend) SetOpRoleBase(opRoleBase *MsgRoleBase) {
	this.opRoleBaseFlag = 1
	this.opRoleBase = opRoleBase
}

func (this *MsgTestSend) GetOpRoleBase() *MsgRoleBase {
	return this.opRoleBase
}

File: pytools/data/data.code.java.server/MsgTestSend.java
package protocol;


import network.Packet;
import network.PacketUtil;

import java.util.List;
import java.util.ArrayList;


public class MsgTestSend
{
	private byte id_u8;
	private MsgRoleBase role_base;
	private List<Float> id_f32 = new ArrayList<Float>();
	private byte id_op_u8_flag;
	private byte id_op_u8;
	private byte op_role_base_flag;
	private MsgRoleBase op_role_base;


	public MsgTestSend()
	{

	}

	public MsgTestSend(Packet packet)
	{
		id_u8 = packet.readByte();
		role_base = new MsgRoleBase(packet);
		int id_f32_count = PacketUtil.readUShort(packet.readShort());
		for (int i = 0; i < id_f32_count; i++)
		{
			id_f32.add(packet.readFloat());
		}
		id_op_u8_flag = packet.readByte();
		if (id_op_u8_flag == 1)
		{
			id_op_u8 = packet.readByte();
		}
		op_role_base_flag = packet.readByte();
		if (op_role_base_flag == 1)
		{
			op_role_base = new MsgRoleBase(packet);
		}
	}

	public byte[] encode()
	{
		Packet packet = new Packet();
		packet.writeByte(id_u8);
		packet.writeBytes(role_base.getBytes());
		int id_f32_count = id_f32.size();
		packet.writeShort((short)id_f32_count);
		for (int i = 0; i < id_f32_count; i++)
		{
			packet.writeFloat((float)id_f32.get(i));
		}
		packet.writeByte(id_op_u8_flag);
		if (id_op_u8_flag == 1)
		{
			packet.writeByte(id_op_u8);
		}
		packet.writeByte(op_role_base_flag);
		if (op_role_base_flag == 1)
		{
			packet.writeBytes(op_role_base.getBytes());
		}
		return packet.readBytes();
	}

	public byte[] getBytes()
	{
		return this.encode();
	}



	public void setIdU8(byte id_u8)
	{
		this.id_u8 = id_u8;
	}
	public int getIdU8()
	{
		return PacketUtil.readUByte(this.id_u8);
	}

	public void setRoleBase(MsgRoleBase role_base)
	{
		this.role_base = role_base;
	}
	public MsgRoleBase getRoleBase()
	{
		return this.role_base;
	}

	public void setIdF32(List<Float> id_f32)
	{
		this.id_f32 = id_f32;
	}
	public List<Float> getIdF32()
	{
		return this.id_f32;
	}

	public void setIdOpU8(byte id_op_u8)
	{
		this.id_op_u8_flag = (byte)1;
		this.id_op_u8 = id_op_u8;
	}
	public int getIdOpU8()
	{
		return PacketUtil.readUByte(this.id_op_u8);
	}

	public void setOpRoleBase(MsgRoleBase op_role_base)
	{
		this.op_role_base_flag = (byte)1;
		this.op_role_base = op_role_base;
	}
	public MsgRoleBase getOpRoleBase()
	{
		return this.op_role_base;
	}

}
